dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/HostConnectionPoolImpl.java

```diff
@@ -359,15 +370,15 @@ public boolean closeConnection(Connection<CL> connection) {
 				conn = availableConnections.poll(duration, unit);
 			} catch (InterruptedException e) {
 				Logger.info("Thread interrupted when waiting on connections");
-				throw new DynoConnectException(e);
+				throw new DynoConnectException(e); // TODO Change this to DynoBorrowedConnectionException(e)
```

Comment:
Decided against the comment; solution doesn't require a specific exception for this but uses PoolExhausted exception instead.